test/react_context_test.dart

```diff
@@ -0,0 +1,159 @@
+@TestOn('browser')
+@JS()
+library react_test_utils_test;
+
+import 'dart:html' as html;
+
+import 'package:js/js.dart';
+import 'package:test/test.dart';
+
+import 'package:react/react.dart' as react;
+import 'package:react/react_client.dart';
+import 'package:react/react_dom.dart' as react_dom;
+
+import 'shared_type_tester.dart';
+
+main() {
+  setClientConfiguration();
+
+  testTypeValue(dynamic typeToTest) {
+    var mountNode = new html.DivElement();
+    var contextTypeRef;
+    var consumerRef;
+    react_dom.render(
+        ContextProviderWrapper({
+          'contextToUse': TestContext,
+          'value': typeToTest,
+        }, [
+          ContextTypeComponent({
+            'ref': (ref) {
+              contextTypeRef = ref;
+            }
+          }),
+          ContextConsumerWrapper({
+            'contextToUse': TestContext,
+            'ref': (ref) {
+              consumerRef = ref;
+            }
+          })
+        ]),
+        mountNode);
+    expect(contextTypeRef.context, same(typeToTest));
+    expect(consumerRef.latestValue, same(typeToTest));
+  }
+
+  group('New Context API (Component2 only)', () {
```

Comment:
Are there any tests that the correct `nextContext` arg is passed into `shouldComponentUpdate`? I can't seem to find any. If not, could you add some?